pallets/democracy/src/lib.rs

```diff
@@ -1266,12 +1267,26 @@ impl<T: Config> Pallet<T> {
 								ensure!(matches!(scope, UnvoteScope::Any), Error::<T>::NoPermission);
 								prior.accumulate(unlock_at, balance)
 							}
+							false
+						} else {
+							let should_lock = if let AccountVote::Standard { vote, .. } = votes[i].1 {
+								let unlock_at = end.saturating_add(
+									T::VoteLockingPeriod::get().saturating_mul(vote.conviction.lock_periods().into()),
+								);
+								let now = frame_system::Pallet::<T>::block_number();
+								now < unlock_at
+							} else {
+								false
+							};
+
+							should_lock
 						}
 					}
-					None => {} // Referendum was cancelled.
-				}
+					None => false, // Referendum was cancelled.
+				};
+
 				votes.remove(i);
```

Comment:
wait, you still remove vote but not lock?

i guess it is always supposed to be reverted, otherwise user could get stuck. can you pls add test that user can in fact remove vote after conviction expires?

Reply:
no, that's not correct. if you dont have staking positions, it behaves like it used to.  meaning, you can always remove your vote.